docs/csharp/programming-guide/concepts/async/index.md

```diff
@@ -132,10 +143,34 @@ Console.WriteLine("Breakfast is ready!");
 
 Another option is to use <xref:System.Threading.Tasks.Task.WhenAny%2A>, which returns a `Task<Task>` that completes when any of its arguments completes. You can await the returned task, knowing that it has already finished. The following code shows how you could use <xref:System.Threading.Tasks.Task.WhenAny%2A> to await the first task to finish and then process its result. After processing the result from the completed task, you remove that completed task from the list of tasks passed to `WhenAny`.
 
-[!code-csharp[AwaitAnyTask](./snippets/index/AsyncBreakfast-final/Program.cs#AwaitAnyTask)]
-
-After all those changes, the final version of `Main` looks like the following code:
+```csharp
+var breakfastTasks = new List<Task> { eggsTask, baconTask, toastTask };
+while (breakfastTasks.Count > 0)
+{
+    Task finishedTask = await Task.WhenAny(breakfastTasks);
+    if (finishedTask == eggsTask)
+    {
+        Console.WriteLine("eggs are ready");
+    }
+    else if (finishedTask == baconTask)
+    {
+        Console.WriteLine("bacon is ready");
+    }
+    else if (finishedTask == toastTask)
+    {
+        Console.WriteLine("toast is ready");
+    }
+    breakfastTasks.Remove(finishedTask);
+}
+```
```

Comment:
Why the snippet reference was changed to code-fence?

Reply:
This was switched to a code-fence so that I had ultimate control over it. In the final version, we now output the entire file and highlight the `Main` function. I didn't want the starter or final versions to have any `// <Snippet*` in them as they would render in the full _Program.cs_ that we now output.